drake/systems/plants/test/testIKMoreConstraints.cpp

```diff
@@ -0,0 +1,221 @@
+  // = model->forwardKin(l_foot_pt, l_foot, 0, 0, 0).value();
+  lfoot_pos0(0) = 0;
+  lfoot_pos0(1) = 0.13;
+  lfoot_pos0(2) = 0.08;
+  Vector3d lfoot_pos_lb = lfoot_pos0;
+  lfoot_pos_lb(0) +=0.001;
+  lfoot_pos_lb(1) +=0.001;
+  lfoot_pos_lb(2) +=0.001;
+  Vector3d lfoot_pos_ub = lfoot_pos_lb;
+  lfoot_pos_ub(2) += 0.01;
+  // std::cout << lfoot_pos0.transpose() << " lfoot\n" ;
+  WorldPositionConstraint* kc_lfoot_pos = new WorldPositionConstraint(model,l_foot,l_foot_pt,lfoot_pos_lb,lfoot_pos_ub,tspan);
+  Eigen::Vector4d quat_des(1,0,0,0);
+  double tol = 0.0017453292519943296;
+  WorldQuatConstraint* kc_lfoot_quat = new  WorldQuatConstraint(model, l_foot, quat_des, tol, tspan);
+
+  // 5 Right Foot Position and Orientation Constraints
+  Vector3d r_foot_pt = Vector3d::Zero();
+  Vector3d rfoot_pos0;
+  //Vector3d rfoot_pos0 = model->forwardKin(r_foot_pt, r_foot, 0, 0, 0).value();
+  rfoot_pos0(0) = 0;
+  rfoot_pos0(1) = -0.13;
+  rfoot_pos0(2) = 0.08;
+  Vector3d rfoot_pos_lb = rfoot_pos0;
+  rfoot_pos_lb(0) +=0.001;
+  rfoot_pos_lb(1) +=0.001;
+  rfoot_pos_lb(2) +=0.001;
+  Vector3d rfoot_pos_ub = rfoot_pos_lb;
+  rfoot_pos_ub(2) += 0.001;
+  //std::cout << rfoot_pos0.transpose() << " lfoot\n" ;
+  WorldPositionConstraint* kc_rfoot_pos = new WorldPositionConstraint(model,r_foot,r_foot_pt,rfoot_pos_lb,rfoot_pos_ub,tspan);
+  WorldQuatConstraint* kc_rfoot_quat = new  WorldQuatConstraint(model, r_foot, quat_des, tol, tspan);
+
+
+  // 6 Right Position Constraints (actual reaching constraint)
+  Vector3d r_hand_pt = Vector3d::Zero();
+  Vector3d rhand_pos0;
+  //Vector3d rhand_pos0 = model->forwardKin(r_hand_pt, r_hand, 0, 0, 0).value();
+  rhand_pos0(0) = 0.2;
+  rhand_pos0(1) = -0.5;
+  rhand_pos0(2) = 0.4;
+
+  Vector3d rhand_pos_lb = rhand_pos0;
+  rhand_pos_lb(0) +=0.05;
+  rhand_pos_lb(1) +=0.05;
+  rhand_pos_lb(2) +=0.05;
+  Vector3d rhand_pos_ub = rhand_pos_lb;
+  rhand_pos_ub(2) += 0.05;
+  //std::cout << rhand_pos_ub.transpose() << " rhand\n" ;
+  WorldPositionConstraint* kc_rhand = new WorldPositionConstraint(model,r_hand,r_hand_pt,rhand_pos_lb,rhand_pos_ub,tspan);
+
+
+  // 7 QuasiStatic Constraints
+  QuasiStaticConstraint* kc_quasi = new QuasiStaticConstraint(model, tspan);
+  kc_quasi->setShrinkFactor(0.2);
+  kc_quasi->setActive(true);
+  Eigen::Matrix3Xd l_foot_pts = Eigen::Matrix3Xd::Zero(3, 4);
+  l_foot_pts <<    -0.0820 ,  -0.0820 ,   0.1780 ,   0.1780   , 0.0624  , -0.0624  ,  0.0624 ,  -0.0624  , -0.0811,   -0.0811  , -0.0811 ,  -0.0811;
+  //std::cout << l_foot_pts <<"\n";
+  kc_quasi->addContact(1, &l_foot, &l_foot_pts);
+  Eigen::Matrix3Xd r_foot_pts = Eigen::Matrix3Xd::Zero(3, 4);
+  r_foot_pts <<    -0.0820 ,  -0.0820 ,   0.1780 ,   0.1780   , 0.0624  , -0.0624  ,  0.0624 ,  -0.0624  , -0.0811,   -0.0811  , -0.0811 ,  -0.0811;
+  kc_quasi->addContact(1, &r_foot, &r_foot_pts);
+
+
+
+  int num_constraints = 9;
+  RigidBodyConstraint** constraint_array = new RigidBodyConstraint*[num_constraints];
```

Comment:
just use a std::vector, and then pass vec.data() into inverseKin (until we clean up the inverseKin interface to take a const std::vector&).